src/pipecat/services/ultravox.py

```diff
@@ -362,18 +368,20 @@ async def _process_audio_buffer(self) -> AsyncGenerator[Frame, None]:
                     await self.stop_processing_metrics()
                     
                     logger.info(f"Generated text: {full_response}")
-                    
                     # Create a transcription frame with the generated text
-                    transcription = full_response.strip()
-                    if transcription:
-                        yield TranscriptionFrame(
-                            text=transcription,
-                            interim_text="",
-                            timestamp=time_now_iso8601()
-                        )
-                    else:
-                        logger.warning("Empty transcription result")
-                        yield ErrorFrame("Empty transcription result")
+                    sentences = []
+                    current = ""
+                    for char in full_response.strip():
+                        current += char
+                        if char in '.!?':
+                            sentences.append(current.strip())
+                            current = ""
+                    if current.strip():  # Add any remaining text
+                        sentences.append(current.strip())
```

Comment:
Probably better to use `match_endofsentence()` otherwise you my cut things like `Dr.` or `Ms.` or `3.14`. It would be something like (untested):

```
current_pos = 0
full_response_end = len(full_response)
while len(ull_response) > 0 :
    index = match_endofsentence(full_response)
    if index == 0:
        break
    sentences.append(full_response[:index])
    full_response = full_response[index:]
```